src/Symfony/Component/DependencyInjection/Attribute/AutowireIterator.php

```diff
@@ -50,6 +50,12 @@ public function __construct(
         foreach ($services as $key => $type) {
             $attributes = [];
 
+            if ($type instanceof Autowire) {
+                $references[$key] = new TypedReference($key, 'mixed', attributes: [$type]);
```

Comment:
Should I throw an exception is the key is numeric?

```php
#[AutowireIterator([new Autowire('%some.parameter%')])]
```

Reply:
Ah, for the iterator, yes. For the locator though?

Reply:
Don't we have some logic in place that turns numeric indexes into service-ids for locator? It's not used here?

Reply:
I'm removing this line, it's not needed